common/persistence/nosql/nosql_execution_store_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package nosql
+
+import (
+	"context"
+	"errors"
+	"github.com/uber/cadence/common/types"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	"github.com/uber/cadence/common/log"
+	"github.com/uber/cadence/common/persistence"
+	"github.com/uber/cadence/common/persistence/nosql/nosqlplugin"
+	"github.com/uber/cadence/service/history/constants"
+)
+
+func TestCreateWorkflowExecution(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockDB := nosqlplugin.NewMockDB(ctrl)
+	logger := log.NewNoop()
+
+	store := &nosqlExecutionStore{
+		shardID:    1,
+		nosqlStore: nosqlStore{logger: logger, db: mockDB},
+	}
+
+	ctx := context.Background()
+	newWorkflowSnapshot := persistence.InternalWorkflowSnapshot{
+		VersionHistories: &persistence.DataBlob{},
+		ExecutionInfo: &persistence.InternalWorkflowExecutionInfo{
+			DomainID:   constants.TestDomainID,
+			WorkflowID: constants.TestWorkflowID,
+			RunID:      constants.TestRunID,
+		},
+	}
+
+	testCases := []struct {
```

Comment:
there's at least 10 code paths/branches in `CreateWorkflowExecution`. let's hit all those cases. If you are using VSCode, enable `Cover On Test Package` and type `cmd+shift+p` and type `Go: Test Package`. This will highlight all the covered lines for you

Reply:
So, should I write cases for all the paths? For now I only included success and failure. I was planning to cover other functions in the part2 diff.

I use goland but can try to do similar check there.

Reply:
makes sense to create multiple diffs until we reach desired level of coverage 👍